### PR TITLE
BaSD-MS-34: Modal

### PR DESCRIPTION
**Pull request by leimeterjoaquin**
# Modal
## 2 componentes created:
-Modal
-ErrorModal
## notes
implemented on all entitites exept for profiles and interviews
## image
image

### props
```
  -titleText => string
  -closeModal => function
  -showModal => boolean(state)
  -actionEntity => function
  -leftButtonText => string
  -rightButtonText => string
  -spanObjectArray => object Array with a title and span. (title is optional)
```
### on entity
```
      <Modal
        showModal={showModal}
        closeModal={closeModal}
        actionEntity={deleteClient}
        selectedId={selectedId}
        titleText="Delete a client"
        spanObjectArray={[
          {
            span: 'are you sure you want to delete this client?'
          }
        ]}
        leftButtonText="delete"
        rightButtonText="cancel"
      />
```

### CODE REVIEW

**Review by MicaQuarleri — CHANGES_REQUESTED** [review]
In the delete of applications, when you when you click delete in the modal button it is deleted from the database but the modal does not disappear

**Review by MicaQuarleri — APPROVED** [review]

File: src/Components/Admins/Form/index.js [hunk 1]
```diff
@@ -102,6 +131,19 @@ const Form = () => {
...
   return (
     <div className={styles.container}>
+      <Modal
+        showModal={showModal}
+        closeModal={closeModal}
+        actionEntity={submit}
+        titleText="Save"
+        spanObjectArray={[
+          {
+            span: 'are you sure you want to save these changes?'
```
Comment (aquilesmariani) [inline src/Components/Admins/Form/index.js:141]: I'd rather starting messages with a capital letter
```suggestion
            span: 'Are you sure you want to save these changes?'
```

File: src/Components/Shared/Modal/index.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  showModal,
+  actionEntity,
+  leftButtonText,
+  rightButtonText,
+  spanObjectArray
+}) => {
+  if (!showModal) {
+    return null;
+  }
+  //TODO: change styles after reset
```
Comment (aquilesmariani) [inline src/Components/Shared/Modal/index.jsx:15]: Just a question, is this still needed?

File: src/Components/Shared/Modal/index.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        <div className={styles.content}>
+          {spanObjectArray.map((span) => (
+            <>
+              {span.title ? <h4 key={span.span + 1}>{span?.title}</h4> : null}
+              <span key={span.span}>{span?.span}</span>
+            </>
+          ))}
+        </div>
+        <div className={styles.buttonContainer}>
+          <button onClick={() => actionEntity()}>{leftButtonText.toUpperCase()}</button>
```
Comment (aquilesmariani) [inline src/Components/Shared/Modal/index.jsx:32]: If you don't have to send values through params you can just do this
```suggestion
          <button onClick={actionEntity}>{leftButtonText.toUpperCase()}</button>
```

**Review by aquilesmariani — CHANGES_REQUESTED** [review]
Looks pretty good!

I think we should rename the form component filenames so they are just index.js since we are already providing the entity name and the Form type of component in the path (e.g /Components/Interviews/Form)

Please refer to the message I left on slack for more info.

File: src/Components/Shared/Modal/index.jsx (hunk 2 above)
Comment (leimeterjoaquin) [reply src/Components/Shared/Modal/index.jsx:15]: Yes I should erase that. Maybe add those changes through a new branch later on.

**Review by aquilesmariani — APPROVED** [review]

**Pull request merged**